pkg/storageincentives/agent.go

```diff
@@ -196,13 +215,13 @@ func (a *Agent) start(blockTime time.Duration, blocksPerRound, blocksPerPhase ui
 		}
 	})
 
-	phaseEvents.On(sample, func(ctx context.Context, _ PhaseType) {
+	phaseEvents.On(sample, func(ctx context.Context, block uint64, _ PhaseType) {
 
 		mtx.Lock()
```

Comment:
instead of passing the block to the events publisher, 
you can have a new `block` var at the top of the start function, similar to what we did with the `round` var